### PR TITLE
Bugfix: compute offsets for vehicle types correctly in LocalSeach

### PR DESCRIPTION
**Pull request by wouterkool**
This PR:

Fixes a bug whiched computed offsets for vehicle types incorrectly in the local search. This is only a problem for heteregeneous problem instances.

- [x] Adds and passes relevant tests.
- [x] Has well-formatted code and documentation.

### CODE REVIEW

**Comment by wouterkool** [comment]
Thanks to https://github.com/joopvdh :)

File: pyvrp/search/tests/test_LocalSearch.py [hunk 1]
```diff
@@ -245,3 +245,23 @@ def test_route_vehicle_types_are_preserved_for_locally_optimal_solutions():
...
+
+    neighbours = compute_neighbours(data)
+    ls = cpp_LocalSearch(data, neighbours)
+    ls.add_node_operator(Exchange10(data))
+    sol = Solution(data, [Route(data, [1, 3], 1), Route(data, [2, 4], 1)])
+    current_cost = cost_evaluator.penalised_cost(sol)
+
+    # The local search should not crash and the result should not be worse
+    improved_sol = ls.search(sol, cost_evaluator)
+    assert_(cost_evaluator.penalised_cost(improved_sol) <= current_cost)
```
Comment (N-Wouda) [inline pyvrp/search/tests/test_LocalSearch.py:267]:
```suggestion
def test_bugfix_vehicle_type_offsets():
    """
    See https://github.com/PyVRP/PyVRP/pull/292 for details. This exercises a
    fix to a bug that would crash local search due to an incorrect internal 
    mapping of vehicle types to route indices if the next vehicle type had
    more vehicles than the previous.
    """
    data = read("data/OkSmall.txt")
    data = make_heterogeneous(data, [VehicleType(10, 1), VehicleType(10, 2)])

    ls = cpp_LocalSearch(data, compute_neighbours(data))
    ls.add_node_operator(Exchange10(data))

    cost_evaluator = CostEvaluator(1, 1)

    current = Solution(data, [Route(data, [1, 3], 1), Route(data, [2, 4], 1)])
    current_cost = cost_evaluator.penalised_cost(current)

    improved = ls.search(current, cost_evaluator)
    improved_cost = cost_evaluator.penalised_cost(improved)

    assert_(improved_cost <= current_cost)
```
Adds a bit more context and spaces the blocks out a bit more clearly, I think.

**Review by N-Wouda — COMMENTED** [review]
Nice catch!

File: pyvrp/search/tests/test_LocalSearch.py [hunk 2]
```diff
@@ -245,3 +245,27 @@ def test_route_vehicle_types_are_preserved_for_locally_optimal_solutions():
     # the solution, especially not change the vehicle types
     further_improved = ls.search(improved, cost_evaluator)
     assert_equal(further_improved, improved)
+
+
+def test_bugfix_vehicle_type_offsets():
+    """
+    See https://github.com/PyVRP/PyVRP/pull/292 for details. This exercises a
+    fix to a bug that would crash local search due to an incorrect internal
```
Comment (N-Wouda) [inline pyvrp/search/tests/test_LocalSearch.py:253]:
```suggestion
    fix to a bug that would crash local search due to an incorrect internal
```

**Review by N-Wouda — APPROVED** [review]
LGTM!

**Comment by wouterkool** [comment]
Thanks for fixing the formatting (next time I'll probably apply your suggestion locally to catch it in the pre-commit)

**Comment by N-Wouda** [comment]
Please squash merge this next time.

**Pull request merged**